xamarin/Xamarin.Forms
# Material Design Renderers for the Material Visual

### Description of Change ###

The PR to show the changes for the Material controls.

### API Changes ###
 
 - `ActivityIndicator`
 - `Button`
 - `Entry`
 - `Frame`
 - `ProgressBar`
 - `Slider` _iOS only_

### Platforms Affected ### 

- iOS
- Android

### Behavioral/Visual Changes ###

 - New Material renderers
    - Most of the renderers are new, but the entry renderers have been tweaked to reuse code. Instead of `EntryRenderer : ViewRenderer<...>`, and then having to duplicate all that code, the base has been extracted out so that `EntryRenderer : EntryRendererBase<FormsEditText>` and the new `MaterialEntryRenderer : EntryRendererBase<MaterialFormsTextInputLayout>`.
 - Refactored the repeated use of `Internals.Registrar.Registered.GetHandlerForObject<IImageSourceHandler>(source)` and the related logic to load images into an extension method `source.GetNativeImageAsync()`  https://github.com/xamarin/Xamarin.Forms/pull/4733/files#diff-5b0ee3b57729867cde314a6ba13fa2ab
- Generalized Entry https://github.com/xamarin/Xamarin.Forms/pull/4733/files#diff-6d85740c77efaa837d7bb69a42102ef5


### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###
- Test Entry control to ensure non material control works
- Ensure changes to Image Loading  on iOS wasn't broken
https://github.com/xamarin/Xamarin.Forms/pull/4733/files#diff-5b0ee3b57729867cde314a6ba13fa2ab 
https://github.com/xamarin/Xamarin.Forms/pull/4733/files#diff-e7e02ce38379deb0c027bd4e04e17265
- Ensure nuspec updates to support v28 are good
- Test changes to iOS Entry Renderer https://github.com/xamarin/Xamarin.Forms/pull/4733/files#diff-b8c74a29a641c60daba89e6908f411ff

### PR Checklist ###
